### PR TITLE
Typos from marvin.cs.uidaho.edu/misspell.html (rework of #1374): A

### PR DESCRIPTION
**Pull request by bl-ue**
For #1949

### CODE REVIEW

File: codespell_lib/data/dictionary.txt [hunk 1]
```diff
@@ -1650,6 +1664,22 @@ ambuguity->ambiguity
 ambulence->ambulance
 ambulences->ambulances
 amdgput->amdgpu
+amealearate->ameliorate
+amealearative->ameliorative
+amealearator->ameliorator
+amealiarate->ameliorate
+amealiarative->ameliorative
+amealiarator->ameliorator
+ameba->amoeba
```
Comment (warmsocks) [inline codespell_lib/data/dictionary.txt:1673]: This is an interesting one.  A couple of things:

1. "ameba" is an uncommon but correct alternate spelling of "amoeba."
2. The pluralization of ameba/amoeba is disputed; in microbiology, some sources state that the plural properly has the Latinate construction "amoebae"(1), although many common encyclopaedia and dictionaries(2) and some textbooks(3) allow "amebas" and "amebae." Wikipedia also cites "amœba."(4)
 
So, while the short/anglicized plural forms are "okay", mixing alternates in the same piece of writing is not.  E.g. one shouldn't use both "amemba" and "amoebas" in the same text, especially a piece of technical writing in which the author is concerned enough to use CodeSpell.

In short, I think these changes will result in consistent, most-correct usages:

Add amebas->amoebae
Add amebae->amoebae
Add amoebas->amoebae
Add amœba->amoeba
Add amœbae->amoebae
Add amœbas->amoebae

References:

(1) https://www.caister.com/highveld/microbiology/what-is-an-amoeba.html
(2) https://www.merriam-webster.com/dictionary/amoeba
(2) https://www.britannica.com/science/amoeba-order
(2) https://www.oxfordlearnersdictionaries.com/us/definition/english/ameba?q=amebas
(2) https://www.lexico.com/en/definition/amoeba
(3) https://www.ncbi.nlm.nih.gov/books/NBK7742/
(4) https://en.wikipedia.org/wiki/Amoeba

File: codespell_lib/data/dictionary.txt [hunk 2]
```diff
@@ -1650,6 +1664,22 @@ ambuguity->ambiguity
 ambulence->ambulance
 ambulences->ambulances
 amdgput->amdgpu
```
Comment (warmsocks) [inline codespell_lib/data/dictionary.txt:1666]: Not sure of the origin of this one, since it isn't from this PR, but it looks to me like it should be... 

amdgput->AMD GPU

... instead.   
?

File: codespell_lib/data/dictionary.txt [hunk 3]
```diff
@@ -1941,11 +2067,14 @@ aperure->aperture
...
 apeture->aperture
 apetures->apertures
+apihelion->aphelion
+apihelions->aphelions
 apilogue->epilogue
 aplha->alpha
 aplication->application
 aplications->applications
 aplied->applied
+aplikay->appliqué
```
Comment (warmsocks) [inline codespell_lib/data/dictionary.txt:2077]: Add applikay->appliqué
Add applique->appliqué
Add aplikays->appliqués
Add applikays->appliqués

File: codespell_lib/data/dictionary.txt [hunk 4]
```diff
@@ -2234,6 +2372,7 @@ aprroximates->approximates
 aprroximation->approximation
 aprroximations->approximations
 aprtment->apartment
+apyoon->uppugn
```
Comment (warmsocks) [inline codespell_lib/data/dictionary.txt:2375]: Uppugn isn't a word.  Replace with apyoon->oppugn
Add apyooned->oppugned
Add apyooning->oppugning
Add apyoons->oppugns

File: codespell_lib/data/dictionary.txt [hunk 5]
```diff
@@ -2835,6 +3002,8 @@ assymmetric->asymmetric
 assymmetrical->asymmetrical
 assymmetries->asymmetries
 assymmetry->asymmetry
+astarisk->asterisk
+astarisks->asterisks
 asterices->asterisks
 asteriks->asterisk, asterisks,
```
Comment (warmsocks) [inline codespell_lib/data/dictionary.txt:3007]: Add asterik->asterisk
